FlashLFQ/Bayesian Protein Quantification/ProteinQuantificationEngine.cs

```diff
@@ -0,0 +1,617 @@
+                            for (int b = 0; b < biorepCount - 1; b++)
+                            {
+                                peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, treatmentCondition, treatmentCondition, b, b + 1));
+                            }
+                        }
+                        else if (PairedSamples)
+                        {
+                            peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, BaseCondition, treatmentCondition, null, null));
+                        }
+                    }
+                    else
+                    {
+                        peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, BaseCondition, treatmentCondition, null, null));
+                    }
+
+                    ProteinQuantificationEngineResult result;
+                    int measurementCount = peptideFoldChanges.SelectMany(p => p.Item2).Count();
+
+                    if (measurementCount > 1)
+                    {
+                        // run the Bayesian analysis
+                        result = RunBayesianProteinQuant(peptideFoldChanges, protein, treatmentCondition,
+                        randomSeedsForEachProtein[protein][0], null, false, 500, 500, out var mus);
+
+                        lock (experimentalNullResults)
+                        {
+                            experimentalNullResults.Add(protein.ProteinGroupName, new List<ProteinQuantificationEngineResult> { result });
+                        }
+                    }
+                    else
+                    {
+                        continue;
+                    }
+                }
+            });
+
+            double stdDev = 0;
+
+            // this is a weird case where there are very few proteins in the data...
+            if (experimentalNullResults.Count < 10)
+            {
+                if (experimentalNullResults.Any())
+                {
+                    stdDev = experimentalNullResults.SelectMany(p => p.Value).Select(p => p.StandardDeviationPointEstimate).Average();
+                }
+                else
+                {
+                    // TODO: not sure what to do here.. there were no protein quant results for this condition
+                    // for now the cutoff is just made to be a fold-change of 1.0, but maybe an exception should be thrown...
+
+                    // the experimental null is 3 * stdDev so the fold-change cutoff will end up being 1.0
+                    stdDev = 1.0 / 3.0;
+
+                    //throw new MzLibException("Could not determine experimental null for condition " + treatmentCondition);
+                }
+            }
+            else
+            {
+                stdDev = experimentalNullResults.SelectMany(p => p.Value).Select(p => p.FoldChangePointEstimate).InterquartileRange() / 1.34896;
+            }
+
+            experimentalNull = 3.0 * stdDev;
```

Comment:
This is the first time "experimentalNull" is used, so there's no reason to initialize it back at ln 497